agent/src/main/java/com/microsoft/applicationinsights/agent/internal/agent/redis/JedisMethodVisitor.java

```diff
@@ -30,9 +30,12 @@
 import org.objectweb.asm.Opcodes;
 import org.objectweb.asm.Type;
 
+
 /**
  * Created by gupele on 8/6/2015.
+ * @deprecated Replaced with JedisMethodVisitorV2
  */
+@Deprecated 
 final class JedisMethodVisitor extends DefaultMethodVisitor {
```

Comment:
@grlima just was thinking about the design decision. Wouldn't it be nice if we implement an interface like MethodVisitorInterface which has methods like getOnEnterMethod() and getOnEnterMethodSignature(). I believe this will provide extensibility and atleast in theory "Favor composition over inherritance" is true. Let me know your thoughts.

Reply:
it's a debatable design discussion and I do think it's a good idea, but since the existing functionality is based on inheritance (there are other methods in the super classes besides the getOn methods I introduced), I don't want to get into it at this time. We can track this and revisit the discussion at a later point.